ocs_ci/deployment/aws.py

```diff
@@ -89,7 +89,13 @@ def check_cluster_existence(self, cluster_name_prefix):
                 False otherwise
 
         """
-        if self.aws.get_instances_by_name_pattern(cluster_name_prefix):
+        instances = self.aws.get_instances_by_name_pattern(cluster_name_prefix)
+        instance_objs = [self.aws.get_ec2_instance(ins.get('id')) for ins in instances]
+        non_terminated_instances = [
+            ins for ins in instance_objs if ins.state
+            .get('Code') != constants.INSTANCE_TERMINATED
+        ]
+        if non_terminated_instances:
```

Comment:
I would rather used an explicit check like `len(non_terminated_instances) > 0` ...

Reply:
You could also consider logging the non terminated instances.

Reply:
Sorry @mbukatov, just now see your comments.
Regarding the first one, how is it different? if non_terminated_instances is basically the same as if the len of it is 0.
Regarding the second one, I'll raise a PR for improving the logging

Reply:
https://github.com/red-hat-storage/ocs-ci/pull/804